src/main/java/com/rog/EShop/services/UserService.java

```diff
@@ -1,34 +1,41 @@
 package com.rog.EShop.services;
 
-import com.rog.EShop.dto.UserDto;
-import com.rog.EShop.dto.UserRegisterDto;
+import com.rog.EShop.dto.*;
 import com.rog.EShop.entity.Role;
 import com.rog.EShop.entity.User;
 import com.rog.EShop.exceptions.BadRequestException;
 import com.rog.EShop.exceptions.ConflictException;
 import com.rog.EShop.exceptions.NotFoundException;
 import com.rog.EShop.mapper.UserMapper;
+import com.rog.EShop.properties.KeycloakProperties;
 import com.rog.EShop.repository.UserRepository;
-import org.springframework.security.core.userdetails.UserDetails;
-import org.springframework.security.core.userdetails.UserDetailsService;
-import org.springframework.security.core.userdetails.UsernameNotFoundException;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.*;
 import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
 
+import java.time.Instant;
 import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 
 @Service
-public class UserService implements UserDetailsService {
+public class UserService {
     private final UserRepository userRepository;
     private final UserMapper userMapper;
-    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+    private static KeycloakProperties keycloakProperties;
```

Comment:
change `private static` to `private final` for `KeycloakProperties`, because you get this bean from spring in constructor. It make no sense to have it "static". Also remove "static" for several methods - those are "private" functions, not "public".